scripts/build-scenario.ts

```diff
@@ -0,0 +1,73 @@
+// Build a scenario with given ID from CLI args
+
+// Usage: `yarn scenario [ID]`
+// ID = name of scenario script OR 'default' if no ID provided
+
+import { ScenarioBuilder, Scenario } from './content-utils'
+import { promises } from 'fs'
+import { GameWorld } from '../src/game/ContentTypes'
+const { writeFile, mkdir } = promises
+
+const id = process.argv.length >= 3 ? process.argv[2] : 'default'
+
+async function buildScenario(id: string) {
+    try {
+        const builder: ScenarioBuilder = (
+            await import(`./scenarios/${id}/index.ts`)
+        ).builder
+
+        const scenario = builder.run()
+
+        console.log(JSON.stringify(scenario, null, 2))
+        console.log(`✅ Sucessfully built scenario "${id}"`)
+
+        return scenario
+    } catch (e) {
+        throw new Error(`Error: Failed to build scenario "${id}" - ${e}`)
+    }
+}
+
+async function exportScenario(id: string, scenario: Scenario) {
+    // TODO: export scenario to JSON
+    // Build into separate files and store in a output directory.
+    // Create files in a directory, or write to existing if they already have been created.
+
+    type Parts = {
+        [K in keyof GameWorld]: string
+    }
+
+    const scenarioParts: Parts = {
+        cards: 'cards',
+        eventCards: 'event-cards',
+        events: 'events',
+        defaultState: 'default-state',
+        stats: 'stats',
+    }
+
+    // Create the necessary directories if they are missing
+    const outputDir = `${__dirname}/build/${id}`
+    await mkdir(outputDir, { recursive: true })
+
+    await Promise.all(
+        Object.entries(scenarioParts).map(([part, fileName]) =>
+            writeFile(
+                `${outputDir}/${fileName}.json`,
+                JSON.stringify(scenario[part as keyof GameWorld], null, 4),
+            ),
+        ),
+    )
+
+    // TODO: improve the export, verify that it works correcly by diffing JSON output with default scenario
```

Comment:
Another way to create tests for scenarios later on, when the game logic has been separated from the component, is to setup jest tests running a number of actions and probing for expected states. (In order to maintain deterministic test results one could inject the random-function upon creation making the experience only truly random when playing the game and completely deterministic when testing)

Reply:
Right, testing would be really useful to solve this! Tests could define exactly how scenarios should work and then ensure they do. This would work very well with a TDD approach to content development and game design - but it would make us further dependent on scripting as the primary way of creating game content.

However that might not be a bad thing, since scripting is more powerful than graphical editors.

Reply:
This will be resolved with #112